lib/ansible/module_utils/network/eos/config/l2_interfaces/l2_interfaces.py

```diff
@@ -207,6 +207,11 @@ def _state_deleted(want, have):
 
 def set_interface(want, have):
     commands = []
+
+    want_mode = want.get("mode")
+    if want_mode and want_mode != have.get("mode"):
```

Comment:
Thanks for the update n If that's the case, I can document the behavior under module documentation, coz otherwise the idempotency will fail in respective scenario.